### DISCUSSION TITLE
some of tailwindss class not working

### DISCUSSION THREAD

**Discussion by arunsingh28** [tailwindlabs/tailwindcss#4498] [Help]
```
const HeaderItem = ({title,Icon}) => {
    return (
        <div>
        <Icon className="h-8" />
        <p>{title}</p>
        </div>
    )
}

export default HeaderItem
```

in this ``` class="h-8" ```  not woking and does't show any class related to **h-8** while inspecting element.
and this thing happen only when i use next.js framework

**Comment by arunsingh28** [comment]
module.exports = {
mode: "jit",
purge: ["./pages/**/*.{js,ts,jsx,tsx}","./components/**/*.{js,ts,jsx,tsx}"],
darkMode: false, // or 'media' or 'class'
theme: {
extend: {},
},
variants: {
extend: {},
},
plugins: [require('tailwind-scrollbar-hide')],
}

**Comment by devntv** [comment]
> ```
> const HeaderItem = ({title,Icon}) => {
>     return (
>         <div>
>         <Icon className="h-8" />
>         <p>{title}</p>
>         </div>
>     )
> }
> 
> export default HeaderItem
> ```
> 
> in this `class="h-8"` not woking and does't show any class related to **h-8** while inspecting element.
> and this thing happen only when i use next.js framework



> ```
> const HeaderItem = ({title,Icon}) => {
>     return (
>         <div>
>         <Icon className="h-8" />
>         <p>{title}</p>
>         </div>
>     )
> }
> 
> export default HeaderItem
> ```
> 
> in this `class="h-8"` not woking and does't show any class related to **h-8** while inspecting element.
> and this thing happen only when i use next.js framework

i think maybe the error is your tailwind settings.
from package.json and  check `dependencies` or `devDependencies` may be missing tailwind (tailwind 2.1.2 -Current versionn)
```"devDependencies": {"autoprefixer": "^10.2.5", "postcss": "^8.2.10","tailwindcss": "^2.1.2" }```
maybe you are missing the code

**Comment by simonswiss** [comment]
Hey!

It's really hard to tell without seeing your code, but my _guess_ is that in your `Icon` component, you're not doing anything with the `className` prop.

If you're calling `<Icon className="p-8" />`, `className becomes a "prop" on the `Icon` component. The `Icon` component needs to _do something_ with it (like merging it with the other classes applied to that component), otherwise it will be completely ignored.

If you have some code to share, I'm happy to provide a bit more assistance than that 👍
